File: 1550.js
// 1550. Three Consecutive Odds
// Given an integer array arr, return true if there are three consecutive odd numbers in the array. Otherwise, return false.

var threeConsecutiveOdds = function(arr) {
    let oddCt = 0
    for (let int of arr) {
        if (int % 2 !== 0) {
            oddCt++
        } else {
            oddCt = 0
        }
        if (oddCt === 3) {
            return true
        }
    }
    return false
};

console.log(threeConsecutiveOdds([2,6,4,1]))
console.log(threeConsecutiveOdds([2,6,4,1,3,5]))

File: 1550.py
# 1550. Three Consecutive Odds
# Given an integer array arr, return true if there are three consecutive odd numbers in the array. Otherwise, return false.

from typing import List

class Solution:
    def threeConsecutiveOdds(self, arr: List[int]) -> bool:
        streak = 0
        for num in arr:
            if num % 2 == 1:
                streak += 1
                if streak == 3:
                    return True
            else:
                streak = 0
        return False

s = Solution()
print(s.threeConsecutiveOdds([2,6,4,1]))
print(s.threeConsecutiveOdds([1,2,34,3,4,5,7,23,12]))